x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -326,15 +330,15 @@ func TestInput(t *testing.T) {
 						"value": `[[index .last_response.body "@timestamp"]]`,
 					},
 				},
-				"request.tracer.filename": "../../logs/httpjson/http-request-trace-*.ndjson",
-				"verifyfilepath":          true,
+				"request.tracer.filename": dir,
```

Comment:
Use `filepath.Join(t.TempDir(), "http-request-trace-*.ndjson")` here so that each test case has its own isolated temp dir that is managed by the `testing` package.

Reply:
Using `os.TempDir()` instead. There seems to be a [problem](https://github.com/golang/go/issues/51442#issuecomment-1507718399) in GoLang 1.20 with removing `t.TempDir()` in Windows platforms

Reply:
Could these be an indicator that our test is not stopping properly and closing the tracer log file? You aren't checking error returned by `os.RemoveAll()` so you might just be ignoring the problem when you use `os.TempDir()`.

Reply:
This was what I was thinking. We never actively close the logger on the basis that it's intended to run forever. We can though, which would make final writes potentially safer. e.g. for cel:
```diff
diff --git a/x-pack/filebeat/input/cel/input.go b/x-pack/filebeat/input/cel/input.go
index 07c20a3680..e0858b752b 100644
--- a/x-pack/filebeat/input/cel/input.go
+++ b/x-pack/filebeat/input/cel/input.go
@@ -663,6 +663,11 @@ func newClient(ctx context.Context, cfg config, log *logp.Logger) (*http.Client,

        if cfg.Resource.Tracer != nil {
                w := zapcore.AddSync(cfg.Resource.Tracer)
+               go func() {
+                       // Close the logger when we are done.
+                       <-ctx.Done()
+                       cfg.Resource.Tracer.Close()
+               }()
                core := ecszap.NewCore(
                        ecszap.NewDefaultEncoderConfig(),
                        w,
```